projects/cashmere/src/lib/scroll-nav/nav/scroll-nav.component.ts

```diff
@@ -0,0 +1,58 @@
+import {Component, ElementRef, ViewEncapsulation, Renderer2, AfterViewInit, QueryList, ContentChildren} from '@angular/core';
+import { ScrollNavLinkDirective } from './scroll-nav-link.directive';
+
+/** Container for scroll navigation links. */
+@Component({
+    selector: 'hc-scroll-nav',
+    encapsulation: ViewEncapsulation.None,
+    styleUrls: ['scroll-nav.component.scss'],
+    templateUrl: 'scroll-nav.component.html'
+})
+export class HcScrollNavComponent implements AfterViewInit {
+    @ContentChildren(ScrollNavLinkDirective) public linkList: QueryList<ScrollNavLinkDirective>;
+    public get _links(): Array<HTMLElement> { return this.linkList.toArray().map(e => e._el.nativeElement); }
+    private readonly ACTIVE_CLASS = 'hc-scroll-nav-active';
+
+    constructor(public _elementRef: ElementRef, private renderer: Renderer2) {}
+
+    public ngAfterViewInit(): void {
+        this._links.forEach((e, i) => {
+            if (i === 0) {
+                e.classList.add(this.ACTIVE_CLASS);
+            }
+            this.renderer.listen(e, 'click', () => {
+                this.onNavClick(e);
+            });
+        });
+    }
+
+    public setActiveClassById(id: string): void {
+        const link = this._links.find(e => e.getAttribute('hcScrollLink') === id);
+        if (!link) {
+            console.warn(`Failed to mark active class. Could not find the element with the data target for id: ${id}.`);
```

Comment:
I would rather throw an error. It seems like this is the sort of thing that would pass QA and become tech debt forever because of a bad setup.